spec/services/preserved_object_handler_spec.rb

```diff
@@ -337,33 +337,27 @@
   end
 
   describe '#with_active_record_transaction_and_rescue' do
-    context 'bogus endpoint' do
-      let(:wrong_ep) do
-        Endpoint.create!(
-          endpoint_name: 'wrong_endpoint',
-          endpoint_type: Endpoint.default_storage_root_endpoint_type,
-          endpoint_node: 'localhost',
-          storage_location: 'blah',
-          recovery_cost: 1
-        )
-      end
-      let(:bad_po_handler) { described_class.new(druid, 6, incoming_size, wrong_ep) }
-
-      before do
-        po = PreservedObject.create!(druid: druid, current_version: 2, preservation_policy: default_prez_policy)
-        PreservedCopy.create!(
-          preserved_object: po, # TODO: see if we got the preserved object that we expected
-          version: po.current_version,
-          size: 1,
-          endpoint: ep,
-          status: Status.default_status
-        )
-      end
-
-      it '#confirm_version rolls back preserved object if the preserved copy cannot be found' do
-        bad_po_handler.confirm_version
-        expect(PreservedObject.find_by(druid: druid).current_version).to eq 2
-      end
+    it '#confirm_version rolls back preserved object if there is a problem updating preserved copy' do
+      wrong_ep = Endpoint.create!(
+        endpoint_name: 'wrong_endpoint',
+        endpoint_type: Endpoint.default_storage_root_endpoint_type,
+        endpoint_node: 'localhost',
+        storage_location: 'blah',
+        recovery_cost: 1
+      )
+      po = PreservedObject.create!(druid: druid, current_version: 2, preservation_policy: default_prez_policy)
+      PreservedCopy.create!(
+        preserved_object: po,
+        version: po.current_version,
+        size: 1,
+        endpoint: wrong_ep,
```

Comment:
i think if you make this just `ep`, then there will be a mismatch between where the copy lives and the endpoint you feed in when creating `bad_po_handler` (where i'd keep `wrong_ep`)... which will result in an error when calling `#confirm_version`, and then i think you could get rid of the explicit `and_raise`.

alternatively, if you'd prefer the explicit `and_raise` on `save!`, i think you could probably get rid of the `wrong_ep` definition and just use `ep` in both places?   i think i have a slight preference for the `wrong_ep` approach, but would certainly be ok with switching to the `and_raise` approach instead.  but pretty sure you don't need both (i'd initially implemented the `wrong_ep` thing to get around doing an `and_raise`).

but i am all for the general thing happening here, with the removal of some unnecessary `context` and `let` stuff.

Reply:
My thoughts were that this ensured that there wasn't already a pres_copy in previous tests, and that raising an error on save! was a way to have crystal clarity in the test how we we triggered the desired situation.   The previous code a) no longer worked with other modifications and b) was not as crystal clear in the test code itself as to what was happening.

Reply:
I actually like the explicit raise here.